tests/components/azure_data_explorer/conftest.py

```diff
@@ -0,0 +1,145 @@
+"""Test fixtures for ADX."""
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    AZURE_DATA_EXPLORER_PATH,
+    BASE_CONFIG_FREE,
+    BASE_CONFIG_FULL,
+    BASIC_OPTIONS,
+)
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@dataclass
+class FilterTest:
+    """Class for capturing a filter test."""
+
+    entity_id: str
+    expected_count: int
+
+
+@pytest.fixture(name="filter_schema")
+def mock_filter_schema():
+    """Return an empty filter."""
+    return {}
+
+
+@pytest.fixture(name="entry_managed")
+async def mock_entry_fixture_managed(hass, filter_schema):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, DOMAIN, {DOMAIN: {CONF_FILTER: filter_schema}}
+    )
+    assert entry.state == ConfigEntryState.LOADED
+
+    # Clear the component_loaded event from the queue.
+    async_fire_time_changed(
+        hass,
+        utcnow() + timedelta(seconds=entry.options[CONF_SEND_INTERVAL]),
+    )
+    await hass.async_block_till_done()
+    return entry
+
+
+@pytest.fixture(name="entry_queued")
+async def mock_entry_fixture_queued(hass, filter_schema):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=BASE_CONFIG_FREE,
```

Comment:
As this and the above fixture are only different in the data, I suggest you create a fixture for the data, which is then passed into one single fixture.

Reply:
I have been looking into this, and I do not know how to do it. the fixture depends on different data. with this we would end up with three fixtures, one with the common code, injected into the two with the different data, in turn injected into the tests? If this is true, I get the idea, but despite I spend some time investigating it, I have no idea on how to implement this.

Reply:
If it is easier for you, you can also create a protected function, which will be called from both fixtures passing the data. So we reduce the duplicate code too

Reply:
This still needs to be resolved. Please don't close conversations without commenting or implementing the suggestion.

Reply:
I belived it was solved, I have created the _entry fixture that is used entry_managed and entry_queued in the effort to reduce the code dublication, Is that not what you requested?